### PR TITLE
(GH-401) Document assembly verification opt-out

### LINKED ISSUE #401: Inventory and document all cake config values
I.e. Settings_SkipVerification is missing.
https://github.com/cake-build/cake/blob/6a606c0665813ee561ef3e4cce5adbd170e2d415/src/Cake.Core/Reflection/AssemblyVerifier.cs#L21

Should be on
https://cakebuild.net/docs/fundamentals/default-configuration-values

So post inventory we hadn't missed that much, there's more settings coming, but we'll do them in a separate pr.

- [ ] Skip Verification
- [ ] ~~NuGet In Process client~~ _this is coming_

| Command line                                   | Environment variable                           |  Config value                                      | Possible values                                   |  Desciption                                     | 
|-----------------------------------------|------------------------------------------|-----------------------------------------|-----------------------------------------|-----------------------------------------|
| --settings_skipverification                 | CAKE_SETTINGS_SKIPVERIFICATION    | [Settings]<br>SkipVerification=        | true / false                                          | Enables to opt-out of the verification of minimum required version Cake.Core for addins. |
| --nuget_useinprocessclient               | CAKE_NUGET_USEINPROCESSCLIENT  | [NuGet]<br>UseInProcessClient=    | true / false                                          | Opt-in to not use nuget.exe for tool/addin installation |

### PR DESCRIPTION
**Pull request by devlead**
* adds documentation for missing Skip Verification
* fixes #401 
* also adds wyam packages.xml artifact to .gitignore

### CODE REVIEW

File: input/docs/fundamentals/default-configuration-values.md [hunk 1]
```diff
@@ -155,3 +155,46 @@ Modules=./tools/Modules
 ```sh
 cake.exe --paths_modules=./tools/Modules
 ```
+
+
+<hr/>
+
+# Skip Verification
+
+If any breaking changes are introduced to Cake, we'll set the minimum supported version of the `Cake.Core` assembly, any addin/assembly that references a older version of `Cake.Core` will generate an exception and ultimately abort the execution of the script. This is to avoid runtime errors during execution of the script.
```
Comment (gep13) [inline input/docs/fundamentals/default-configuration-values.md]: `an older`

File: input/docs/fundamentals/default-configuration-values.md [hunk 2]
```diff
@@ -155,3 +155,46 @@ Modules=./tools/Modules
...
+
+But if you really need to use an addin/assembly and verified that the breaking change introduced doesn't concern the addin/assembly directly, i.e. it doesn't use a removed property or changed interface, for this scenario we'll allow you to temporarily opt-out of the assembly verification until addin/assembly author has released a version of the assembly targeting the minimum required version of `Cake.Core`.
```
Comment (gep13) [inline input/docs/fundamentals/default-configuration-values.md]: `and have verified`

File: input/docs/fundamentals/default-configuration-values.md [hunk 3]
```diff
@@ -155,3 +155,46 @@ Modules=./tools/Modules
...
+
+A word of **caution**, this setting is global for all addins/assemblies your script references and should therefor be considered a temporary quick fix.
```
Comment (gep13) [inline input/docs/fundamentals/default-configuration-values.md]: `therefore`

**Comment by devlead** [comment]
@gep13 feedback addressed.

**Pull request merged**